src/GithubApi.hs

```diff
@@ -4,17 +4,18 @@
 -- Licensed under the Apache License, Version 2.0 (the "License");
 -- you may not use this file except in compliance with the License.
 -- A copy of the License has been included in the root of the repository.
-
-{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DataKinds #-}
```

Comment:
The rewrite of this file (and quite a few others) is much the same as `Git.hs`, so the changes can be glossed over. I'll add comments for noteworthy things.